### PR TITLE
[INT-161] Initial version of the library

### PR DESCRIPTION
**Pull request by gromakovsky**
I am creating a new tiny library. Haddock documentation describes the problem solved by this library. I am going to write a short blog-post about it as well. It's ready for review now, I just need to update the blog post link when the blog post is ready.

Note that CI reports some broken links, I think it's fine. They will be alive when we merge this PR, upload this package to Hackage and Stackage.

### CODE REVIEW

File: src/Control/Exception/Uncaught.hs [hunk 1]
```diff
@@ -3,10 +3,107 @@
...
+-- be applied to the body of @main@.
+--
+-- It works similarly to 'displayUncaughtException', but instead of
+-- catching and throwing exceptions it modifies the uncaught exception
+-- handler to wrap the exception before processing it.
+-- When the action finishes, the uncaught exception handler
+-- is restored (normally it should not matter because the function is
+-- supposed to be applied to the whole @main@).
+-- The handler won't be restored in case of exception (because otherwise
+-- it wouldn't work).
```
Comment (Martoon-00) [inline src/Control/Exception/Uncaught.hs]: Could you elaborate a bit more on this last sentence?

Comment (gromakovsky) [reply src/Control/Exception/Uncaught.hs]: Done, see the new commit.

**Review by Martoon-00 — APPROVED** [review]

**Comment by gromakovsky** [comment]
@Martoon-00 I have pushed some changes, please check. Something else should be changed in the article as well, but I'd like to make sure that current version of the library is ok first. Will proceed to the article afterwards.

**Comment by Martoon-00** [comment]
> @Martoon-00 I have pushed some changes, please check. Something else should be changed in the article as well, but I'd like to make sure that current version of the library is ok first. Will proceed to the article afterward.

Those changes look good to me :+1:

**Comment by gromakovsky** [comment]
Ok, I am going to squash fixups soon then.

**Pull request merged**